MibTeX/src/de/mibtex/export/ExportTypo3Bibtex.java

```diff
@@ -0,0 +1,153 @@
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import de.mibtex.BibtexViewer;
+import de.mibtex.export.typo3.Filters;
+import de.mibtex.export.typo3.Modifiers;
+import de.mibtex.export.typo3.Typo3Entry;
+
+/**
+ * Exports the bibtex file to bibtex in a carefully adjusted format such that the BibTex-Importer of Typo3 (Website-Framework) can read it correctly.
+ *
+ * @author Paul Bittner
+ */
+public class ExportTypo3Bibtex extends Export {
+	// TODO: Having numbers in Bibtex tags is not conforming the Bibtex standard.
+	public final static String Typo3TagsAttribute = "typo3Tags";
+	private final static String MYabrv = "MYabrv.bib";
+	private final static String MYshort = "MYshort.bib";
+	// We do not consider MYfull because it is deprecated.
+	
+	/**
+	 * Select the filter you need to export only the publications you are interested in.
+	 * A Typo3Entry t gets selected if bibFilter.test(t) returns 'true'.
+	 * You may use or compose default filters and helper functions from de.mibtex.export.typo3.Filters.
+	 * For instance, if you want to select a subset of specific publications manually, use
+	 *     Filters.KeyIsOneOf("Key1", "Key2", ...)
+	 * to select all publications with these keys.
+	 */
+	private final Predicate<Typo3Entry> bibFilter =
+			//Filters.Any()
+			//Filters.KeyIsOneOf("RSC+:SE21", "KJN+:SE21", "KJN+:FASE20")
+			Filters.WithThomasAtUlm
+	;
+	
+	/**
+	 * Select the modifiers you want to apply to each entry after filtering.
+	 * Each modifier is a function taking a Typo3Entry and returning the modified Typo3Entry.
+	 * Default implementations can be found in de.mibtex.export.typo3.Modifiers.
+	 * For instance, Modifiers::MarkIfThomasIsEditor checks if Thomas Th�m is an editor of the entry and if so,
+	 * adds a specific tag to the publication that we use for the website.
+	 * Some modifiers are dedicated to resolving duplicate entries (w.r.t. titles) because Typo3 considers entries with the same title to be the same.
+	 * If unsure, leave unchanged.
+	 */
+	private final List<Function<Typo3Entry, Typo3Entry>> modifiers = Arrays.asList(
+			  Modifiers.MarkIfThomasIsEditor
+			, Modifiers.MarkIfToAppear
+			
+			// Resolving duplicates
+			, Modifiers.IfKeyIs("useRLB+:AOSD14", Modifiers.MarkIfTechreport)
+			, Modifiers.IfKeyIs("TKL:SPLC18", Modifiers.AppendToTitle("(Second Edition)"))
+			, Modifiers.IfKeyIs("KAT:TR16", Modifiers.MarkIfTechreport)
+			, Modifiers.IfKeyIs("TKK+:SPLC19", Modifiers.MarkAsExtendedAbstract)
+			, Modifiers.IfKeyIs("KTS+:SE19", Modifiers.MarkAsExtendedAbstract)
+			, Modifiers.IfKeyIs("KJN+:SE21", Modifiers.MarkAsExtendedAbstract)
+			, Modifiers.IfKeyIs("RSC+:SE21", Modifiers.MarkAsExtendedAbstract)
+			, Modifiers.IfKeyIs("KTP+:SE19", Modifiers.MarkAsExtendedAbstract)
+	);
+	
+	
+	public ExportTypo3Bibtex(String path, String file) throws Exception {
+		super(path, file);
+	}
+
+	@Override
+	public void writeDocument() {
```

Comment:
Split into several methods

Reply:
I extracted the step that checks for duplicate entries and issues a warning.
I did not extract anything else as I could not find any lines of code for which it would be reasonable to extract them to their own method without introducing more boilerplate (in terms of LOC) than the initial code itself.

Reply:
I also extracted writing files in UTF8 now which I should have done right away.